### ISSUE TITLE
Make Webhook compatible with Discord

### ISSUE THREAD

**Issue by kzaoaai** [boredazfcuk/docker-icloudpd#92]
The current body format does not work with Discord, and returns "Cannot send an empty message".
It's a very minor fix to make it work with the already existing webhook integration.

I replaced "data" with "content" in line 536:
` --data "{ \"data\" : \"${2}\" }" \` to ` --data "{ \"content\" : \"${2}\" }" \`
and it works now.

Worth mentioning port 433 and webhook_path=/api/webhook**s**/ should be used.

Complete working configuration (after modifying line 536:

`   --env notification_type=Webhook \

   --env webhook_server=discord.com \

   --env webhook_port=443 \

   --env webhook_path=/api/webhooks/ \

   --env webhook_id=#############/<redacted>\

   --env webhook_https=True \`

**Comment by jmengit** [comment]
Really, the "data" in "{ \"data\" should be a user set variable. For example, IFTTT uses "value1" as its webhook payload variable. This simple change would make webhooks compatible with MANY more services.

**Comment by jmengit** [comment]
Pretty sure if line 536 is changed to the following the program could be run with a "webhook_keyword" variable and would work as described above. Not terribly familiar with shell scripts though so not actually submitting a pull request...

 --data "{ \\"${webhook_keyword=data}\\" : \\"${2}\\" }" \\

**Comment by kzaoaai** [comment]
I agree. My proposal was specific for my use case, didn't realize other platforms use other "values". Sounds like the ENVIRONMENT VARIABLE route is the way to go. Never done a pull request, just pointing this potential fix out.

I created a pull request after going through the script **(my first one ever)**. I tested it by replacing the script and setting the environment variable and it works as expected.
I couldn't build a test docker image as the build process kept failing with the provided dockerfile.

**Comment by boredazfcuk** [comment]
I’ll leave webhook as it is and I’ll make Discord it’s own notification type. I have a Discord account I can test with.

The Webhook stuff was requested by someone for a Home Automation system which I can’t test, so I’d rather leave that untouched. I don’t want to change something and break it for them.

I’m on holiday at minute and back mid-week, so will be a while before I can implement this.

**Comment by kzaoaai** [comment]
It's for homeassistant, a great software which I use, just not for this specific purpose.

Thanks for update. Cheers!

**Comment by jmengit** [comment]
@boredazfcuk I REALLY recommend making this a user defined variable that simply defaults to "data". This will mean it doesnt impact your HomeAssistant user AT ALL but also allows the webhooks notifier to be used by a LOT more services. As it stands you almost might as well call the the webhook notifier "HomeAssistant notifier" because thats pretty much the only service the uses "data" as the first JSON POST property.

@kzaoaai Pull request looks good and I tested it with HomeAssistant, Discord, and IFTTT. BUT I would probably echo the variable on line 126 to match the syntax of the all other variables being ingested.

**Comment by kzaoaai** [comment]
@jmengit added to PR.
